### ISSUE TITLE
Wire:model array does not works well with wire:key

### ISSUE THREAD

**Issue by xitox97** [livewire/livewire#1357]
### Description
Firstly, I'm not sure whether what I'm doing is the correct way or not.

I have an issue where wire:key added in same element that has wire:model. The issue are when I delete or remove array value of the wire:model, it will remove for awhile and the data will come back for other method that is called.

### Steps to reproduce
I added [Laravel playground](https://laravelplayground.com/#/snippets/dada73e6-56c6-4289-9e08-6df0f51e81a5)

1. Click add + button
2. Fill the value at new row input
3. Click remove button for new added row
4. Try edit second row with any value or you can click **add +** button again
5. The removed value will display again.

p.s If I **dd** to see the value of **$this->numbers** after click remove button it will just show **2 value**, however if I **dd** inside **total()** method it will still showed the removed value.

**Context**
- Livewire version: v1.3.2
- Laravel version: 7.4
- Browser: Opera

**Solution**
Currently I fix it by remove wire:keyup, and added call **total() function** inside **hydrate function**
This is the [playground ](https://laravelplayground.com/#/snippets/f2cd53b0-c919-4b04-94ee-d75fc197c24d )that I fixed it.

**Comment by calebporzio** [comment]
Thank you for submitting.

I tried to diagnose the problem for a little bit using the playground and couldn't pin-point it.

I think there are too many moving parts in this issue to properly diagnose.

Please strip down the code to the most granular level to help pin-point the problem.

Thanks!

**Comment by neamtua** [comment]
Hi,

I will open this again because I've ran into the same issue today. I'm using the latest Livewire version.

Here's a stripped down version of my code. 
https://laravelplayground.com/#/snippets/77329702-d4a6-44cc-9314-02ee8069e95c

How to reproduce:
1. Press the Add button 3 times until you have 3 rows in the table.
2. Enter values in all the fields so that keyup is triggered.
3. Delete second row.
4. Delete last row.

What will happen:
The second row will be deleted. The last row will only reset the value instead of deleting it.
If you try and type in the first row, a third row will appear.

I've spent hours debugging this until i've realised that the issue is the keyup event.
